lib/eventmachine.rb

```diff
@@ -881,6 +881,19 @@ def self.open_datagram_socket address, port, handler=nil, *args
     c
   end
 
+  # Same as open_datagram_socket but also multicast subscribes
+  def self.open_datagram_socket_and_mcastjoin address, port, handler=nil, *args
+
+    klass = klass_from_handler(Connection, handler, *args)
+    s = EM::open_udp_socket address, port.to_i
+    ip =  IPAddr.new( address ).hton + IPAddr.new("0.0.0.0").hton
+    set_sock_opt s, Socket::IPPROTO_IP, Socket::IP_ADD_MEMBERSHIP, ip
```

Comment:
I'm a little worried about this creating a precedent for adding separate top-level methods for each sockopt you might want to set. Could you do the IP_ADD_MEMBERSHIP in the post_init of your connection handler class?